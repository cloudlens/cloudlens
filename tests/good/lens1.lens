lens propLens () {
     {
        print("\n--- New Session ---");
        sessionTime = 0;
    }

    match {
       	"Process (?<process>\w+)";
        "execution time (?<time>\d+)s# time:int"
    }
    
    stream (entry in log) {
        print(entry.process + ": " + entry.time);
        sessionTime += entry.time;
    }
    
     {
        print("\n- Session Time: " + sessionTime + "\n");
    }
    
    stream (entry in log) when (entry.time) {
        entry.prop = (entry.time / sessionTime) * 100;
    }
    
    stream (entry in log) {
        var prop = entry.prop;
        print(entry.process + ": " + prop.toFixed(2) + "%");
    }
}
    
match {
    "Session (?<date>\d\d-\d\d-\d\d\d\d)"   
}

group (entry) upon entry.date in session {
    "^"
}

stream (entry) when (entry){
    propLens(entry.session)
}
