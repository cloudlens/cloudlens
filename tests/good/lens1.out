
--- New Session ---
emacs: 123
eclipse: 4000
cloudlens: 2
firefox: 6899

- Session Time: 11024

emacs: 1.12%
eclipse: 36.28%
cloudlens: 0.02%
firefox: 62.58%

--- New Session ---
mail: 3233
firefox: 12049

- Session Time: 15282

mail: 21.16%
firefox: 78.84%

--- New Session ---
cloudlens: 66
firefox: 8762
eclipse: 2390
cloudlens: 2
cloudlens: 5

- Session Time: 11225

cloudlens: 0.59%
firefox: 78.06%
eclipse: 21.29%
cloudlens: 0.02%
cloudlens: 0.04%
